OpenRA.Mods.Common/Traits/BotModules/BotModuleLogic/BaseBuilderQueueManager.cs

```diff
@@ -134,7 +134,9 @@ bool TickQueue(IBot bot, ProductionQueue queue)
 				// HACK: HACK HACK HACK
 				// TODO: Derive this from BuildingCommonNames instead
 				var type = BuildingType.Building;
-				if (world.Map.Rules.Actors[currentBuilding.Item].HasTraitInfo<AttackBaseInfo>())
+
+				// Check if Building is a defense and if we should place it towards the enemy or not.
+				if (world.Map.Rules.Actors[currentBuilding.Item].HasTraitInfo<AttackBaseInfo>() && world.LocalRandom.Next(100) < baseBuilder.Info.PlaceDefenseTowardsEnemyChance)
 					type = BuildingType.Defense;
```

Comment:
I wonder if we want to rename this now.

Reply:
I also find it a bit strange that the `PlaceDefenseTowardsEnemyChance` parameter dictates a `BuildingType`. We discussed it on IRC and pchote suggested renaming the enum to `PlacementType` or similar.